bc_gym_planning_env/envs/base/reward.py

```diff
@@ -132,40 +141,53 @@ def reward(self, state):
         :param state State: full state of the environment
         :return float: the reward
         """
-        if self._state.done():
-            return 0.0
-
-        # See if you have reached any new point
-        last_reached_idx = find_last_reached(
-            state.pose,
-            self._state.path,
-            self._params.spatial_precision,
-            self._params.angular_precision
-        )
-
-        if last_reached_idx is not None and last_reached_idx >= self._state.target_idx:
-            # assign the reward for reaching a waypoint
-            next_idx = last_reached_idx + 1
-
-            self._state.target_idx = next_idx
-
-            # set up the new waypoint
-            if not self._state.done():
-                dist_to_goal, _ = pose_distances(self._state.current_goal_pose(), state.pose)
-                self._state.min_spat_dist_so_far = dist_to_goal
-            else:
-                self._state.min_spat_dist_so_far = 0.0
-
-            return 1.0
-        else:
-            # maybe we get some progress towards current waypoint
-            dist_to_goal, _ = pose_distances(self._state.current_goal_pose(), state.pose)
-
-            if dist_to_goal < self._state.min_spat_dist_so_far:
-                # We progressed toward the current waypoint
-                reward = self._state.min_spat_dist_so_far - dist_to_goal
-                self._state.min_spat_dist_so_far = dist_to_goal
-                return reward * self._params.spatial_progress_multiplier
-            else:
-                # We didn't do any progress
-                return 0.0
+
```

Comment:
I need a different reward structure. Can I directly change this reward class or you prefer me create another class?

Reply:
What is the reward structure you need? Let's write new reward provider class.

Reply:
just as the changes I made here. I'll create a new class to put it there